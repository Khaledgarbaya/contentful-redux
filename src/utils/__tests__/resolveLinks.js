const { resolveLinksMutating } = require('../resolveLinks');

describe('contentful link resolval', () => {

	const entryWithEntry = {
		entryField: {
			sys: {
				type: 'Link',
				linkType: 'Entry',
				id: 'e2'
			}
		},
		sys: {
			type: 'Entry',
			id: 'e1'
		}
	};

	const entryWithAsset = {
		assetField: {
			sys: {
				type: 'Link',
				linkType: 'Asset',
				id: 'a1'
			}
		},
		sys: {
			type: 'Entry',
			id: 'e2'
		}
	};

	it('should resolve links to entries', () => {
		const entries = {
			e1: JSON.parse(JSON.stringify(entryWithEntry)),
			e2: JSON.parse(JSON.stringify(entryWithAsset))
		};
		expect(resolveLinksMutating([entries.e1], { ...entries })).toMatchSnapshot();
	});

	it('should resolve links to assets', () => {
		const asset = {
			sys: {
				type: 'Asset',
				id: 'a1'
			},
			myProperty: 'Hello World'
		};
		const assets = {
			a1: asset
		};
		const entries = {
			e2: JSON.parse(JSON.stringify(entryWithAsset))
		};
		expect(resolveLinksMutating([entries.e2], { ...entries, ...assets })).toMatchSnapshot();
	});

});
