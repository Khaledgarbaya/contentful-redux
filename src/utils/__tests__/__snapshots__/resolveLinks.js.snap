// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contentful link resolval should resolve links to assets 1`] = `
Array [
  Object {
    "assetField": Object {
      "myProperty": "Hello World",
      "sys": Object {
        "id": "a1",
        "type": "Asset",
      },
    },
    "sys": Object {
      "id": "e2",
      "type": "Entry",
    },
  },
]
`;

exports[`contentful link resolval should resolve links to entries 1`] = `
Array [
  Object {
    "entryField": Object {
      "assetField": Object {
        "sys": Object {
          "id": "a1",
          "linkType": "Asset",
          "type": "Link",
        },
      },
      "sys": Object {
        "id": "e2",
        "type": "Entry",
      },
    },
    "sys": Object {
      "id": "e1",
      "type": "Entry",
    },
  },
]
`;
